File: go/pull-with-push-override/fs/foo-sfl.go
/*
 *  Copyright © 2022 Paulo Villela. All rights reserved.
 *  Use of this source code is governed by the MIT license
 *  that can be found in the LICENSE file.
 */

package fs

import (
	"fmt"
	"github.com/pvillela/module-config/go/pull-with-push-override/config"
)

type FooSflCfgInfo struct {
	X string
}

type FooSflCfgSrc = config.CfgSrc[FooSflCfgInfo]

var FooSflCfgSrcV = config.MakeCfgSrc[FooSflCfgInfo](FooSflCfgAdapter)

func FooSfl() {
	fmt.Println(FooSflCfgSrcV().X)
	BarBf()
}

File: src/main/kotlin/pullcfgwithoverride/fs/fooSfl.kt
/*
 *  Copyright © 2022 Paulo Villela. All rights reserved.
 *  Use of this source code is governed by the MIT license
 *  that can be found in the LICENSE file.
 */

package pullcfgwithoverride.fs

import pullcfgwithoverride.config.CfgSrc
import pullcfgwithoverride.config.makeCfgSrc

data class FooSflCfgInfo(
	val x: String
)

var fooSflCfgSrc: CfgSrc<FooSflCfgInfo> = makeCfgSrc(::fooSflCfgAdapter)

fun fooSfl() {
	println(fooSflCfgSrc().x)
	barBf()
}
